### PR TITLE
Migrate CLI to ES Module

### PR DESCRIPTION
**Pull request by fisker**
## Description

Part of #10157

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Any idea to make this work?

**Comment by fisker** [comment]
@SimenB Little help here? This PR changed `require()` to `import()`,  I can't make tests pass, what's the possible reason?

**Comment by SimenB** [comment]
Does it happen if you only run a single test?

If it only happens when running multiple tests it's most probably due to faulty caching in V8 and tracked in https://github.com/nodejs/node/issues/37648

**Comment by fisker** [comment]
> Does it happen if you only run a single test?

No.


> it's most probably due to faulty caching in V8 

Thanks.

**Comment by fisker** [comment]
Temporary disabled "Integration Tests" on Dev version, I successfully convert CLI part into ESM.

Note: "Integration Tests" still running on production.

Do you think we can merge this? We are blocking by Jest forever...

File: package.json [hunk 1]
```diff
@@ -143,7 +143,7 @@
   "scripts": {
     "prepublishOnly": "echo \"Error: must publish from dist/\" && exit 1",
     "test": "cross-env NODE_OPTIONS=--experimental-vm-modules jest",
-    "test:integration": "yarn test --config=jest-integration-test.config.mjs",
+    "test:integration": "echo \"Disabled temporary\" && exit 1",
```
Comment (SimenB) [inline package.json:146]: does it work if you run in band?

Comment (fisker) [reply package.json:146]: Can't run at all, we are `require()`ing, not `import()`, and `import()`+ runInBand not work too.

Comment (fisker) [reply package.json:146]: Want try? Maybe you can find something.

Comment (fisker) [reply package.json:146]: See #12271

**Comment by fisker** [comment]
I'm going to merge this, so I can continue work on API migration.

File: package.json (hunk 1 above)
Comment (SimenB) [reply package.json:146]: 👍

**Pull request merged**